### PR TITLE
Fix text direction nullability

### PR DESCRIPTION
**Pull request by dnfield**
See https://github.com/flutter/flutter/pull/111493#discussion_r970021924

### CODE REVIEW

**Review by LongCatIsLooong — APPROVED** [review]

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 1]
```diff
@@ -223,10 +223,13 @@ class TextPainter {
   /// parameters, lays it out with the supplied [minWidth] and [maxWidth], and
   /// returns its [TextPainter.width] making sure to dispose the underlying
   /// resources.
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/text_painter.dart:225]: we should add something about how this is expensive

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 2]
```diff
@@ -223,10 +223,13 @@ class TextPainter {
...
+  ///
+  /// Unlike the [TextPainter] constructor, the [textDirection] parameter is
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/text_painter.dart:227]: The grammar in this sentence is dubious (it implies "the constructor is not required"). In general I'm not sure as written that this paragraph really helps the reader understand what's happening.

I would instead say something like "The [text] and [textDirection] are required as they have no reasonable defaults." or even maybe just not say anything at all.

File: packages/flutter/lib/src/painting/text_painter.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter/lib/src/painting/text_painter.dart]: Done

File: packages/flutter/lib/src/painting/text_painter.dart (hunk 2 above)
Comment (dnfield) [reply packages/flutter/lib/src/painting/text_painter.dart:229]: Removed.

**Comment by Hixie** [comment]
test-exempt: api change with no logic change

**Pull request merged**